webapp/src/app/core/services/block.service.ts

```diff
@@ -16,4 +19,20 @@ export class BlockService {
     delete this.blockMap[block.id];
     this.refreshChanges.next();
   }
+
+  addBlock(blockType: BlockType): void {
+    const newBlock = new BlockData(randstr64(10), blockType);
+    newBlock.order = this.blockCount++;
+    newBlock.variableId = randCustomString(numerals, 4);
+    this.blockMap[newBlock.id] = newBlock;
+  }
+
+  getBlockList(): BlockData[] {
+    const blockList: BlockData[] = Object.values(this.blockMap);
+    blockList.sort((a, b) => Math.sign(a.order - b.order)); // Oldest first
+    // [0,1,3,4,5] -> [0,1,2,3,4]
+    // In case some block were deleted.
+    blockList.forEach((block, i) => block.order = i);
```

Comment:
Seems to me that, this creates a situation where `blockList` and `blockMap` are different in `order`.
So when we call `updateBlock(block)`, it will have `blockList` order, while the rest of the blocks have `blockMap` order.

I think we can keep the numbers with gaps.
Alternatively, change it also in `blockMap`.

Reply:
> Alternatively, change it also in blockMap
We do. We don't change `blockList`. We change each element of `blockList`. The elements are the same, which `blockMap` has. So order is refreshed in `blockMap` too.

Reply:
Right..